docs/development.txt

```diff
@@ -0,0 +1,84 @@
+they may request that a feature branch be created in JanusGraph repository.  If the developers are not committers, a
+JanusGraph committer will be assigned as the shepard for that branch.  The shepard will be responsible for merging
+pull requests to that branch.
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+Pull Requests
+-------------
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC - review-then-commit). This workflow is
+performed using Github's pull request process. In instances where the committer deems the full process
+unnecessary, commit-then-review (CTR) may be employed. This should be used infrequently and only for extremely trivial
+changes. Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.  The committer
+should also send a message to https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] to notify other
+project members of the CTR.  If another committer responds with a -1, the commit should be rolled back and the formal
+RTC process should be followed.
+
+Users wishing to contribute to JanusGraph should fork the https://github.com/janusgraph/janusgraph[JanusGraph]
+repository and then submit pull requests using the Github pull request process. Please include the relevant issue
+number in the title of your pull request. A vote (-1, 0, +1) is required before a pull request may be merged.
+Non-committers are free, and encouraged to vote. Their votes will be non-binding, but can be taken into consideration
+and are still very valuable community input. The following rules apply to the voting process.
+
+* Three +1 votes are required merge a pull request
+* One or more -1 votes will veto the pull request until the noted issue(s) are addressed and the -1 vote is withdrawn
+* -1 votes will not be considered valid without an explanation
+* Committers may vote on their own pull requests
+
+[[release-policy]]
+Release Policy
+--------------
+
+Any JanusGraph committer may propose a release. To propose a release, simple start a new RELEASE thread on
+https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] proposing the new release and requesting
+feedback on what should be included in the release. After consensus is reached the release manager will perform
+the following tasks:
+
+* Create a release branch so that work may continue on `master`
+* Prepare the release artifacts
+* Call a vote to approve the release on https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev]
+
+Committers will be given 72 hours to review and vote on the release artifacts. A minimum of three +1 votes and more
++1's than -1's are required for a release to be approved.
```

Comment:
@ptgoetz I know this was just a suggestion, you okay with me switching this back to a -1 vetoing a release?  I'm on the fence with this one because in many cases, I would imagine if anyone felt there was a serious enough issue to warrant a veto, we'd probably be in agreement, but maybe not...